### PR TITLE
Use BaseItemKind where possible

### PR DESCRIPTION
**Pull request by crobibero**
Got a little big...

### CODE REVIEW

File: tests/Jellyfin.Api.Tests/Helpers/RequestHelpersTests.cs [hunk 1]
```diff
@@ -59,31 +59,7 @@ public static void GetOrderBy_Success(IReadOnlyList<string> sortBy, IReadOnlyLis
...
-                "AggregateFolder",
-                "Audio",
-                "BasePluginFolder",
-                "CollectionFolder"
-            };
-
-            var res = RequestHelpers.GetItemTypeStrings(input);
-
-            Assert.Equal(expected, res);
+            Assert.Empty(Array.Empty<BaseItemKind>());
```
Comment (nielsvanvelzen) [inline tests/Jellyfin.Api.Tests/Helpers/RequestHelpersTests.cs:62]: Isn't this test useless now?

**Review by nielsvanvelzen — APPROVED** [review]
API changes look fine!

File: Jellyfin.Api/Controllers/LibraryController.cs [hunk 2]
```diff
@@ -715,26 +716,26 @@ public async Task<ActionResult> GetDownload([FromRoute, Required] Guid itemId)
...
-                includeItemTypes.Add(nameof(Series));
+                includeItemTypes.Add(BaseItemKind.Series);
             }
             else
             {
                 // For non series and movie types these columns are typically null
                 isSeries = null;
                 isMovie = null;
-                includeItemTypes.Add(item.GetType().Name);
+                includeItemTypes.Add(item.GetBaseItemKind());
```
Comment (cvium) [inline Jellyfin.Api/Controllers/LibraryController.cs:738]: Now that GetBaseItemKind is used so often, can we cache the value? Enum.Parse isn't cheap.

**Review by 1337joe — APPROVED** [review]

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 3]
```diff
@@ -196,57 +196,56 @@ public class SqliteItemRepository : BaseSqliteRepository, IItemRepository
...
         {
-            "Audio",
-            "MusicAlbum",
-            "MusicVideo",
-            "AudioBook",
-            "AudioPodcast"
+            BaseItemKind.Audio,
+            BaseItemKind.MusicAlbum,
+            BaseItemKind.MusicVideo,
+            BaseItemKind.AudioBook
```
Comment (Bond-009) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs:248]: I think `HashSet` is the best collection type for this now, plain arrays should be fine

**Review by Bond-009 — APPROVED** [review]

**Comment by 1337joe** [comment]
Doing this on insert:

https://github.com/jellyfin/jellyfin/blob/b5d4fdb56e59772b925d75cfe1c8ba691dbe1e77/Emby.Server.Implementations/Data/SqliteItemRepository.cs#L718

is incompatible with doing this on query (when `includeTypes` contains `BaseItemKind`):

https://github.com/jellyfin/jellyfin/blob/b5d4fdb56e59772b925d75cfe1c8ba691dbe1e77/Emby.Server.Implementations/Data/SqliteItemRepository.cs#L3584

Will also need some migration work to convert existing installs to use `BaseItemKind` as the database type.

**Pull request merged**